### PR TITLE
Add not present condition for content-type

### PR DESCRIPTION
**Pull request by smallhive**
I'm not sure I added it correctly

### CODE REVIEW

File: bearer/bearer.go [hunk 1]
```diff
@@ -52,6 +52,7 @@ func (b *Generator) NewBearer(email string, currentEpoch uint64, msPerEpoch int6
 	rec.AddFilter(eacl.HeaderFromObject, eacl.MatchStringNotEqual, object.AttributeContentType, "text/html")
 	rec.AddFilter(eacl.HeaderFromObject, eacl.MatchStringNotEqual, object.AttributeContentType, "text/htmlh")
 	rec.AddFilter(eacl.HeaderFromObject, eacl.MatchStringNotEqual, object.AttributeContentType, "")
+	rec.AddFilter(eacl.HeaderFromObject, eacl.MatchNotPresent, object.AttributeContentType, "")
```
Comment (roman-khimov) [inline bearer/bearer.go:55]: Looks like it should be inverted. We're conjugating things to allow here and we're allowing an object if ContentType is present (and not allowing if it's missing).

Comment (smallhive) [reply bearer/bearer.go:55]: Is it possible to invert condifition? I thought it wasn't possible, thus there are `eacl.MatchStringNotEqual` and opposite `eacl.MatchStringEqual`.
There is no `MatchPresent` option. Any hidden options?

Comment (roman-khimov) [reply bearer/bearer.go:55]: It can be expressed as another rule (DENY if not present), but I'm not sure about filter inversion. @carpawell?

Comment (carpawell) [reply bearer/bearer.go:55]: yes, you should deny objects that `MatchNotPresent` if you want to skip objects without `AttributeContentType`. and yes, we can think about `MatchPresent` too (currently there is no such one)

**Comment by roman-khimov** [comment]
@smallhive?

**Comment by smallhive** [comment]
I thought I did. Updated now

File: bearer/bearer.go [hunk 2]
```diff
@@ -63,6 +63,11 @@ func (b *Generator) NewBearer(email string, currentEpoch uint64, msPerEpoch int6
 	eacl.AddFormedTarget(rec2, eacl.RoleOthers)
 	t.AddRecord(rec2)
 
+	rec3 := eacl.CreateRecord(eacl.ActionDeny, eacl.OperationPut)
```
Comment (roman-khimov) [inline bearer/bearer.go:66]: It has to come _before_ `rec`. If `rec` matches this is an unreachable rule.

Comment (smallhive) [reply bearer/bearer.go]: Moved

File: bearer/bearer.go [hunk 3]
```diff
@@ -43,8 +43,13 @@ func (b *Generator) NewBearer(email string, currentEpoch uint64, msPerEpoch int6
...
 
 	t := eacl.CreateTable(b.config.ContainerID)
+	rec := eacl.CreateRecord(eacl.ActionDeny, eacl.OperationPut)
+	rec.AddFilter(eacl.HeaderFromObject, eacl.MatchNotPresent, object.AttributeContentType, "")
+	t.AddRecord(rec)
+	eacl.AddFormedTarget(rec, eacl.RoleOthers)
+
 	// order of rec is important
-	rec := eacl.CreateRecord(eacl.ActionAllow, eacl.OperationPut)
+	rec = eacl.CreateRecord(eacl.ActionAllow, eacl.OperationPut)
```
Comment (roman-khimov) [inline bearer/bearer.go:52]: This will mangle the old `rec`.

File: bearer/bearer.go [hunk 4]
```diff
@@ -43,8 +43,13 @@ func (b *Generator) NewBearer(email string, currentEpoch uint64, msPerEpoch int6
 	epochs := uint64(b.config.ObjectMaxLifetime.Milliseconds() / msPerEpoch)
 
 	t := eacl.CreateTable(b.config.ContainerID)
+	rec := eacl.CreateRecord(eacl.ActionDeny, eacl.OperationPut)
+	rec.AddFilter(eacl.HeaderFromObject, eacl.MatchNotPresent, object.AttributeContentType, "")
+	t.AddRecord(rec)
+	eacl.AddFormedTarget(rec, eacl.RoleOthers)
```
Comment (roman-khimov) [inline bearer/bearer.go:49]: Better have a new record.

File: bearer/bearer.go (hunk 3 above)
Comment (smallhive) [reply bearer/bearer.go]: Updated

File: bearer/bearer.go (hunk 4 above)
Comment (smallhive) [reply bearer/bearer.go]: Refactored in the separate commit, to be able to easily manage order and rules in the future

**Review by roman-khimov — APPROVED** [review]

**Pull request merged**